NumSet/NumSet.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+
+namespace Assingment4.Tests
+{
+    [TestClass]
+    public class NumSetTests
+    {
+        [TestMethod]
+        public void Constructor_PassesParamsArray_AssignsParamsArrayToHashSet()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            HashSet<int> temp = new HashSet<int>(intArray);
+
+            //Act
+            NumSet numSet = new NumSet(intArray);
+
+            //Assert
+            Assert.IsNotNull(numSet.IntSet);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Constructor_PassedNullArgs_ThrowsException()
+        {
+            var numSet = new NumSet(null);
+        }
+
+        [TestMethod]
+        public void ReturnArray_TakesLocalHashSet_ReturnsAnIntArray()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            NumSet numSet = new NumSet(intArray);
+
+            //Act
+            int[] returnArray = numSet.ReturnArray();
+
+            //Assert
+            CollectionAssert.AreEqual(intArray, returnArray);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ReturnArray_HashSetHasNotBeenSetupByConstructor_ThrowsException()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 3, 2 };
+            NumSet numSet = new NumSet(intArray);
+            numSet.IntSet = null;
+
+            //Act
+            numSet.ReturnArray();
+
+            //Assert
+
+        }
+
+        [TestMethod]
+        public void Equals_PassedTwoDifferentNumSet_ReturnsFalse()
+        {
+            //Assign
+            int[] intArray1 = new int[] { 1, 2, 3 };
+            int[] intArray2 = new int[] { 1, 2, 4 };
+            NumSet numSet1 = new NumSet(intArray1);
+            NumSet numSet2 = new NumSet(intArray2);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(numSet1.Equals(numSet2));
+        }
+
+        [TestMethod]
+        public void Equals_PassedTheSameNumSet_ReturnsTrue()
+        {
+            //Assign
+            int[] intArray = new int[] { 1, 2, 3 };
+            NumSet numSet = new NumSet(intArray);
```

Comment:
You'll want to set up a second `NumSet` object rather than comparing the same object to its self.
```suggestion
            NumSet numSet1 = new NumSet(intArray);
            NumSet numSet2 = new NumSet(intArray);
```